tracer/src/Datadog.Trace/ClrProfiler/ServerlessInstrumentation/AWS/LambdaCommon.cs

```diff
@@ -168,5 +188,20 @@ private static void WriteRequestPayload(WebRequest request, string data)
             dataStream.Write(byteArray, 0, byteArray.Length);
             dataStream.Close();
         }
+
+        private static void WriteRequestHeaders(WebRequest request, IDictionary<string, string> context)
+        {
+            foreach (KeyValuePair<string, string> kv in context)
+            {
+                request.Headers.Add(kv.Key, kv.Value);
+            }
+        }
+
+        private static object GetTraceContext(object obj)
```

Comment:
Duck typing is strongly typed, so I think you can return `IDictionary<string, string>` without any casts. 
```suggestion
        private static IDictionary<string, string> GetTraceContext(object obj)
```